client/mainmenu/CCampaignScreen.cpp

```diff
@@ -154,3 +220,52 @@ void CCampaignScreen::CCampaignButton::hover(bool on)
 			hoverLabel->setText(" ");
 	}
 }
+
+void CCampaignScreen::switchPage(int delta, const std::string & campaignSet)
+{
+	currentPage += delta;
+	currentPage = std::clamp(currentPage, 0, maxPages - 1);
+
+	const auto& campaignConfig = CMainMenuConfig::get().getCampaigns();
+
+	updateCampaignButtons(campaignConfig, campaignSet);
+}
+
+void CCampaignScreen::updateCampaignButtons(const JsonNode & parentConfig, const std::string & campaignSet)
```

Comment:
Why you need `const std::string & campaignSet`? From what I see, this class already has this value as a member. Remove?
Same question to `CCampaignScreen::switchPage`

Reply:
Not needed, removed from all places now.